File: data/readable_format/fortran/file1726.f90
program factorialTestProgram
    implicit none
    integer :: result

    call testFactorialCalculation(0, 1)
    call testFactorialCalculation(1, 1)
    call testFactorialCalculation(5, 120)
end program factorialTestProgram

subroutine calculateFactorial(n, result)
    integer, intent(in) :: n
    integer, intent(out) :: result
    result = 1
    do i = 2, n
        result = result * i
    end do
end subroutine calculateFactorial

subroutine testFactorialCalculation(n, expected)
    integer, intent(in) :: n, expected
    integer :: result
    call calculateFactorial(n, result)
    if (result == expected) then
        print *, "Test passed for n =", n
    else
        print *, "Test failed for n =", n, "Expected:", expected, "Got:", result
    endif
end subroutine testFactorialCalculation

File: data/readable_format/cpp/file1726.cpp
#include <iostream>

int calculateFactorial(int n) {
    int result = 1;
    for(int i = 2; i <= n; ++i) {
        result *= i;
    }
    return result;
}

int main() {
    int n = 5;
    std::cout << "The factorial of " << n << " is " << calculateFactorial(n) << std::endl;
    return 0;
}